tests/test_validate_call.py

```diff
@@ -848,6 +848,82 @@ def find_max_validate_return[T](args: Iterable[T]) -> T:
             find_max(1)
 
 
+@pytest.mark.skipif(sys.version_info < (3, 12), reason='requires Python 3.12+ for PEP 695 syntax with generics')
+def test_pep695_with_class():
+    """Primarily to ensure that the syntax is accepted and doesn't raise a NameError."""
+
+    for import_annotations in ('from __future__ import annotations', ''):
+        globs = {}
+        exec(
+            f"""
+{import_annotations}
+from pydantic import validate_call
+
+class A[T]:
+    @validate_call(validate_return=True)
+    def f(self, a: T) -> T:
+        return str(a)
+
+    """,
+            globs,
+        )
+
+        A = globs['A']
+        a = A[int]()
+        # these two are undesired behavior, but it's what happens now
+        assert a.f(1) == '1'
+        assert a.f('1') == '1'
```

Comment:
Because the `validare_return` should convert the returned value back to `int`. This should be equivalent to 
```py
@validate_call(validate_return=True)
def f(x: int)->int:
    return str(x)

print(type(f(1)))  # int
```

Reply:
Ah yes, overlooked the flag. Thanks!